clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractFetch.java

```diff
@@ -97,31 +81,13 @@ public AbstractFetch(final LogContext logContext,
         this.subscriptions = subscriptions;
         this.fetchConfig = fetchConfig;
         this.decompressionBufferSupplier = BufferSupplier.create();
-        this.completedFetches = new ConcurrentLinkedQueue<>();
+        this.fetchBuffer = new FetchBuffer<>(logContext);
         this.sessionHandlers = new HashMap<>();
         this.nodesWithPendingFetchRequests = new HashSet<>();
         this.metricsManager = metricsManager;
         this.time = time;
     }
 
-    /**
```

Comment:
Just wondering, is there a reason to move these 2 to the child class Fetcher? I see they just perform some checks on the what is now the fetcherBuffer, which is defined here in the AbstractFetcher, so I would say they still make sense here, and we would avoid an unneeded change.